packages/vite-plugin-svelte/src/index.ts

```diff
@@ -40,155 +46,164 @@ export function svelte(inlineOptions?: Partial<Options>): Plugin {
 	/* eslint-enable no-unused-vars */
 
 	let resolvedSvelteSSR: Promise<PartialResolvedId | null>;
+	const api: PluginAPI = {};
+	const plugins: Plugin[] = [
+		{
+			name: 'vite-plugin-svelte',
+			// make sure our resolver runs before vite internal resolver to resolve svelte field correctly
+			enforce: 'pre',
+			api,
+			async config(config, configEnv): Promise<Partial<UserConfig>> {
+				// setup logger
+				if (process.env.DEBUG) {
+					log.setLevel('debug');
+				} else if (config.logLevel) {
+					log.setLevel(config.logLevel);
+				}
+				// @ts-expect-error temporarily lend the options variable until fixed in configResolved
+				options = await preResolveOptions(inlineOptions, config, configEnv);
+				// extra vite config
+				const extraViteConfig = buildExtraViteConfig(options, config);
+				log.debug('additional vite config', extraViteConfig);
+				return extraViteConfig;
+			},
 
-	return {
-		name: 'vite-plugin-svelte',
-		// make sure our resolver runs before vite internal resolver to resolve svelte field correctly
-		enforce: 'pre',
-		async config(config, configEnv): Promise<Partial<UserConfig>> {
-			// setup logger
-			if (process.env.DEBUG) {
-				log.setLevel('debug');
-			} else if (config.logLevel) {
-				log.setLevel(config.logLevel);
-			}
-			// @ts-expect-error temporarily lend the options variable until fixed in configResolved
-			options = await preResolveOptions(inlineOptions, config, configEnv);
-			// extra vite config
-			const extraViteConfig = buildExtraViteConfig(options, config);
-			log.debug('additional vite config', extraViteConfig);
-			return extraViteConfig;
-		},
-
-		async configResolved(config) {
-			options = resolveOptions(options, config);
-			patchResolvedViteConfig(config, options);
-			requestParser = buildIdParser(options);
-			compileSvelte = createCompileSvelte(options);
-			viteConfig = config;
-			log.debug('resolved options', options);
-		},
+			async configResolved(config) {
+				options = resolveOptions(options, config);
+				patchResolvedViteConfig(config, options);
+				requestParser = buildIdParser(options);
+				compileSvelte = createCompileSvelte(options);
+				viteConfig = config;
+				// TODO deep clone to avoid mutability from outside?
+				api.options = options;
```

Comment:
this allows other plugins to consume the resolved options. Do we want to use the api namespace or find another way?